src/packet/public_key_encrypted_session_key.js

```diff
@@ -67,13 +72,36 @@ class PublicKeyEncryptedSessionKeyPacket {
    * @param {Uint8Array} bytes - Payload of a tag 1 packet
    */
   read(bytes) {
-    this.version = bytes[0];
-    if (this.version !== VERSION) {
+    let offset = 0;
+    this.version = bytes[offset++];
+    if (this.version !== 3 && this.version !== 6) {
       throw new UnsupportedError(`Version ${this.version} of the PKESK packet is unsupported.`);
     }
-    this.publicKeyID.read(bytes.subarray(1, bytes.length));
-    this.publicKeyAlgorithm = bytes[9];
-    this.encrypted = crypto.parseEncSessionKeyParams(this.publicKeyAlgorithm, bytes.subarray(10));
+    if (this.version === 6) {
+      const sizeFields = bytes[offset++];
+      if (sizeFields !== 0) {
+        this.publicKeyVersion = bytes[offset++];
+        const fingerprintLength = this.publicKeyVersion ? (this.publicKeyVersion >= 5 ? 32 : 20) : 0;
```

Comment:
```suggestion
        const fingerprintLength = this.publicKeyVersion >= 5 ? 32 : 20;
```
public key version of 0 has been removed in favour of the explicit size

Reply:
and we could actually use the `sizeFields - 1` value instead of hardcoding the fingerprint length